core/utils/validate.go

```diff
@@ -23,7 +21,7 @@ type Validated interface {
 
 // Validate returns any errors from calling Validated.ValidateConfig on cfg and any nested types that implement Validated.
 func Validate(cfg interface{}) (err error) {
```

Comment:
How about a more specific name if this is going to live in a general utils package?
```suggestion
func ValidateConfig(cfg interface{}) (err error) {
```

Reply:
If anything, I would suggest making it less, rather than more, specific. That way this can be used generally to validate any struct.

eg.
```
func Validate(s interface{}) error {
...
}

type Validated interface {
     IsValid() error
}
```

Reply:
But isn't the code hyper-specific to toml config? I'm concerned about moving any pieces over to utils:

> Move config utilities relating to TOML, ValidateConfig, into utils.

It would not be appropriate to use most (all?) of these as-is from other parts of the system - TOML especially, since we use a different module here than for job specs, for example. I think they need to remain config scoped, at least by name, but package separation seems cleaner.

Reply:
Hmmm... yes, fair point. How about a configutils package containing DecodeTOML, ValidateConfig and the various errors that might be emitted from ValidateConfig() functions?

What I want to achieve with this is to break the dependency between config/v2 and evmconfig/v2, since previously evmconfig/v2 would depend on config/v2, and this caused some import cycles, by moving the common parts out to a separate package.

Reply:
Actually, in the future LOOP Plugins will be responsible for validating their own config, so we will want to move these generic pieces over to chainlink-relay to be available for other repos. With that in mind, I think it would be simplest to keep a specialized package in the meantime.

Reply:
> Hmmm... yes, fair point. How about a configutils package containing DecodeTOML, ValidateConfig and the various errors that might be emitted from ValidateConfig() functions?

Yeah I think that would work :+1:

Reply:
There is a minimal `package config` in chainlink-relay with copies of some error types:
https://github.com/smartcontractkit/chainlink-relay/tree/main/pkg/config

Reply:
Understood! Would you like me to move these pieces to chainlink-relay and import them here? Or should I just stick to putting them in configutils for now?

Reply:
I think a local package is fine for now. We can move it later when we move the chain family types for LOOPs.